docs/Concepts/Transactions/Trace-Types.md

```diff
@@ -0,0 +1,148 @@
+    },
+    "pc":72,
+    "sub":null
+    },
+    ...
+  ]
+}
+```
+
+| Key       | Value                                                                                |
+|-----------| --------------------------------------------------------------------------------------|
+| `code`    | Code executed by the EVM.  
+| `ops`     | Sequence of EVM operations (opcodes) executed in the transaction.                
+| `cost`    | Gas cost of the opcode. Includes memory expansion costs but not gas refunds. For precompiled contract calls, reports only the actual cost. 
+| `ex`      | Executed operations. 
+| `mem`     | Memory read or written by the operation. 
+| `push`    | Adjusted stack items. For swap, includes all intermediate values and end result. Otherwise, is the value pushed onto stack. 
+| `store`   | Account storage written by the operation.  
+| `used`    | Remaining gas taking into account the all but 1/64th rule for calls. 
+| `pc`      | Program counter.
+| `sub`     | Sub call operations. 
+
+Besu only reports actual data returned from a `RETURN` opcode. Besu does not 
+return the contents of the reserved output space for the call operations. As a result: 
+
+* Besu reports null when a call operation ends because of a `STOP`, `HALT`, `REVERT`, running out of 
+instructions, or any exceptional halts.
+* When a `RETURN` operation returns data of a different length to the space reserved by the call, only
+the data passed to the `RETURN` operation is reported. Besu does not include pre-existing memory data
+or trim the returned data.
+
+For out of gas operations, Besu reports the operation that caused the out of gas exception including
+the calculated gas cost. No `ex` values are reported because the operation is not executed. 
+
+## stateDiff
+
+State changes in the requested block for each transaction represented as 
+a map of accounts to an object. The balance, code, nonce, and storage changes are enumerated
+from immediately before the transaction to after the transaction.  For the `key:value` pairs:  
+
+* `+` indicates the field didn’t exist before and how has the specified value
+* `-` indicates the value was deleted
+* `*` has a from and a to value.
+
+An absent value is distinct from zero when accounts or storage are created or cleared.
+
+```json tab="stateDiff Example" 
+"stateDiff":{
+  "0xfe3b557e8fb62b89f4916b721be55ceb828dbd73":{ 
+    "balance":{ 
+      "*":{ 
+        "from":"0xffffffffffffffffffffffffffffffffc3e12a20b",
+        "to":"0xffffffffffffffffffffffffffffffffc3dc5f091"
+      }
+    },
+    "code":"=",
+    "nonce":{ 
+      "*":{ 
+        "from":"0x14",
+        "to":"0x15"
+      }
+    },
+    "storage":{              
+    }
+  }
+}
+``` 
+
+| Key            | Value                                                                                |
+|-----------     | --------------------------------------------------------------------------------------|
+| `balance`      | Change of balance event   
```

Comment:
The descriptions in the previous table ended with fullstops (.).

Reply:
Added